homeassistant/components/unifi/switch.py

```diff
@@ -37,51 +46,224 @@
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.restore_state import RestoreEntity
 
-from .const import (
-    ATTR_MANUFACTURER,
-    BLOCK_SWITCH,
-    DOMAIN as UNIFI_DOMAIN,
-    DPI_SWITCH,
-    OUTLET_SWITCH,
-    POE_SWITCH,
-)
+from .const import ATTR_MANUFACTURER, DOMAIN as UNIFI_DOMAIN, POE_SWITCH
 from .controller import UniFiController
 from .unifi_client import UniFiClient
 
 CLIENT_BLOCKED = (EventKey.WIRED_CLIENT_BLOCKED, EventKey.WIRELESS_CLIENT_BLOCKED)
 CLIENT_UNBLOCKED = (EventKey.WIRED_CLIENT_UNBLOCKED, EventKey.WIRELESS_CLIENT_UNBLOCKED)
 
-T = TypeVar("T")
+Data = TypeVar("Data")
+Handler = TypeVar("Handler")
+
+Subscription = Callable[[CallbackType, ItemEvent], UnsubscribeType]
+
+
+def async_dpi_group_is_on_fn(
+    api: aiounifi.Controller, dpi_group: DPIRestrictionGroup
+) -> bool:
+    """Calculate if all apps are enabled."""
+    return all(
+        api.dpi_apps[app_id].enabled
+        for app_id in dpi_group.dpiapp_ids or []
+        if app_id in api.dpi_apps
+    )
+
+
+def async_sub_device_available_fn(controller: UniFiController, obj_id: str) -> bool:
+    """Check if sub device object is disabled."""
+    device_id = obj_id.split("_", 1)[0]
```

Comment:
This could use `partition` (see #81493):
```suggestion
    device_id = obj_id.partition("_")[0]
```